### ISSUE TITLE
alpha legend keys darken when geom_ribbons share an alpha value

### ISSUE THREAD

**Issue by twest820** [tidyverse/ggplot2#4677]
One `geom_ribbon(aes(alpha = foo))` is OK with `scale_alpha_manual()`. Two or more ribbons with `alpha = foo` alters the legend keys. Is there a way to get ggplot 3.3.5 to draw the alpha legend correctly? `override.aes` doesn't work here.
```r
# insert reprex here
data = tibble(x = c(0, 1), confInt1Q025 = 0.025, confInt1Q25 = 0.25, confInt1Q75 = 0.75, confInt1Q975 = 0.975,
              confInt2Q025 = confInt1Q025 + x, confInt2Q25 = confInt1Q25 + x, confInt2Q75 = confInt1Q75 + x, confInt2Q975 = confInt1Q975 + x)
# behaves as expected
ggplot(data) +
  geom_ribbon(aes(x = x, ymin = confInt1Q025, ymax = confInt1Q975, alpha = "95% interval", fill = "confInt1")) +
  geom_ribbon(aes(x = x, ymin = confInt1Q25, ymax = confInt1Q75, alpha = "50% interval", fill = "confInt1")) +
  scale_alpha_manual(breaks = c("95% interval", "50% interval"), values = c(0.1, 0.5))
# produces overly dark alpha legend
ggplot(data) +
  geom_ribbon(aes(x = x, ymin = confInt1Q025, ymax = confInt1Q975, alpha = "95% interval", fill = "confInt1")) +
  geom_ribbon(aes(x = x, ymin = confInt1Q25, ymax = confInt1Q75, alpha = "50% interval", fill = "confInt1")) +
  geom_ribbon(aes(x = x, ymin = confInt2Q025, ymax = confInt2Q975, alpha = "95% interval", fill = "confInt2")) +
  geom_ribbon(aes(x = x, ymin = confInt2Q25, ymax = confInt2Q75, alpha = "50% interval", fill = "confInt2")) +
  guides(alpha = guide_legend(override.aes = list(alpha = c(0.1, 0.5)))) + # has no effect, which isn't surprising but was worth a try
  scale_alpha_manual(breaks = c("95% interval", "50% interval"), values = c(0.1, 0.5))
```
The first plot, where the alpha keys correspond adequately well to the shading in the plot:
![image](https://user-images.githubusercontent.com/9359341/142503153-87b992ea-0c28-4673-845c-84e2ec48a8bd.png)
The second plot, where the alpha keys are have darkened even though they should be showing the exact same alpha values as in the first plot:
![image](https://user-images.githubusercontent.com/9359341/142503179-36809ea0-9bdd-45c9-93a0-1141ffac4dd1.png)
The more ribbons the worse this gets. Since I have plots with 5+ overlapping confidence ranges shown this way the alpha keys get so dark they've basically nothing to do with the (correctly) displayed transparency of the ribbons. This reproduces across AGG, Cairo, and Windows so it doesn't look like a graphics device issue.

The least bad workaround I've identified so at the moment is kludgy but probably still better than manual fixup with an image editor (it's a lot messier to implement with the real tibbles than in the repex here).
```R
library(tidyr)
dataWorkaround = data %>% 
  pivot_longer(cols = starts_with("confInt"), names_to = c("confInt", "quantile"), names_pattern = "confInt([[:digit:]])(.*)") %>%
  pivot_wider(names_from = quantile, values_from = value)
ggplot(dataWorkaround) +
  geom_ribbon(aes(x = x, ymin = Q025, ymax = Q975, alpha = "95% interval", fill = confInt)) +
  geom_ribbon(aes(x = x, ymin = Q25, ymax = Q75, alpha = "50% interval", fill = confInt)) +
  guides(alpha = guide_legend(override.aes = list(alpha = c(0.1, 0.5)))) +
  scale_alpha_manual(breaks = c("95% interval", "50% interval"), values = c(0.1, 0.5))
```

**Comment by smouksassi** [comment]
this will always be the case when you add layers as opposed to having one geom/stat call that will draw all your data.
the legend will be maybe you can trick lineribbon to do what you want
https://mjskay.github.io/ggdist/articles/lineribbon.html

**Comment by twest820** [comment]
Thank you, @smouksassi. I wasn't aware of `geom_lineribbon()` and will have to write some code to check. From [its documentation](https://mjskay.github.io/ggdist/articles/lineribbon.html#multiple-lineribbons-on-one-plot-1) the main apparent difference is it's aware of tibble groups and therefore allows slightly simpler code, It's unclear if the lineribbon family uses ggdist's slab alpha scale instead of the regular alpha scale but perhaps slab alpha has useful properties here.

I'm trying but failing to think of a common scenario where multiple geoms would align so closely it would be more natural to show merged rather than individual alpha levels in the legend key. If it's an intentional design choice offering some way to opt out seems reasonable. But no `scale_alpha_discrete(overlay = FALSE)` type option is apparent in the ggplot documentation.

**Comment by twest820** [comment]
A couple minor follow ups:

1. Experimentation suggests this issue also applies to the relevant portions of ggdist. From what I can tell `geom_lineribbon()` does not employ slab alpha. But, as that's the first and only time so far I've tried using ggdist, perhaps I missed something.
2. I've encountered some related limitations of color, fill, and alpha legend merging. Since separate values of `override.aes = list(alpha)` aren't supported for merged color and fill legends you must also merge the alpha legend if you don't want the color legend's segment grobs to be washed out. If limits are used to filter the keys appearing in the merged legend then colors get lost and ribbons render in opaque grey. So it looks like this issue may be an aspect of a more complex one regarding the linkage between ribbons and the alpha scale.

I haven't constructed a repex for the second item as the translation from my actual data and plotting is involved.
